### PR TITLE
port parshah code from C

### PR DESCRIPTION
**Pull request by yparitcher**
this is a port of the parshah code from my C hebrew date & zmanim program [libzmanim](https://github.com/yparitcher/libzmanim)

### CODE REVIEW

**Comment by KosherJava** [comment]
@yparitcher I know that it has been a long time. I had planned on refactoring things, but I see that will delay this too much, and I can easily do it after this is added. One small change request I have is the parsha names in the enums. Can you use the Ashkenazi transliteration as used throughout the codebase. Specifically see https://github.com/KosherJava/zmanim/commit/5c14a07f617fe990238db0fef0114bea5eb22356#diff-70152126b02d93ec0863fe69964ff69b
Thanks

File: src/net/sourceforge/zmanim/hebrewcalendar/JewishCalendar.java [hunk 1]
```diff
@@ -519,6 +667,26 @@ public boolean isRoshChodesh() {
 		return (getJewishDayOfMonth() == 1 && getJewishMonth() != TISHREI) || getJewishDayOfMonth() == 30;
 	}
 
+	/**
+	 * Returns if the day is Shabbos and sunday is Rosh Chodesh.
+	 *
+	 * @return true if it is Shabbos and sunday is Rosh Chodesh.
+	 */
+	public boolean ismacharchodesh()
```
Comment (KosherJava) [inline src/net/sourceforge/zmanim/hebrewcalendar/JewishCalendar.java]: Please use camelCase isMacharChodesh()

File: src/net/sourceforge/zmanim/hebrewcalendar/JewishCalendar.java [hunk 2]
```diff
@@ -519,6 +667,26 @@ public boolean isRoshChodesh() {
...
+	{
+		return (getDayOfWeek() == 7 && (getJewishDayOfMonth() == 30 || getJewishDayOfMonth() == 29));
+	}
+
+	/**
+	 * Returns if the day is Shabbos Mevorchim.
+	 *
+	 * @return true if it is Shabbos Mevorchim.
+	 */
+	public boolean isshabbosmevorchim()
```
Comment (KosherJava) [inline src/net/sourceforge/zmanim/hebrewcalendar/JewishCalendar.java]: Please use camelCase isShabbosMevorchim()

File: src/net/sourceforge/zmanim/hebrewcalendar/JewishCalendar.java [hunk 3]
```diff
@@ -188,6 +210,132 @@ public boolean getInIsrael() {
...
+		}
+		return parshah.NONE;
+	}
+
+	/**
+	 * returns a parshah enum if the week is one of the four parshahs if it is Shabbos.
+	 * returns NONE if a weekday
+	 * @return one of the four parshahs
+	*/
+	public parshah getspecialshabbos()
```
Comment (KosherJava) [inline src/net/sourceforge/zmanim/hebrewcalendar/JewishCalendar.java]: Please use camelCase getSpecialShabbos()

**Comment by yparitcher** [comment]
fixed

**Pull request merged**